Sources/Apodini/Semantic Model Builder/Model/EncodeOptionalEndpointParameter.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  File.swift
+//  
+//
+//  Created by Nityananda on 07.01.21.
+//
+
+protocol EncodeOptionalEndpointParameter {
+    func createParameterWithWrappedType(
+        name: String,
+        label: String,
+        necessity: EndpointParameterNecessity
+    ) -> AnyEndpointParameter
+}
+
+// MARK: - Parameter+EncodeOptionalEndpointParameter
+
+extension Parameter: EncodeOptionalEndpointParameter where Element: ApodiniOptional, Element.Member: Codable {
```

Comment:
The `EncodeOptionalEndpointParameter` and its extension to `Parameter` are really only a hacky way to get to the Wrapped type, if the generic type of `Parameter` is a `Optional`. It is only used during Model Building and I don't consider that a "project internal API".

I would suggest to make it private, so it isn't visible to the other files in the project, which would require it being move to the ModeBuilder itself. Similar to the refactoring I have done in one of my latest PRs: https://github.com/Apodini/Apodini/blob/806528af4cc6d192e5ab4f527389d221d39bcc12/Sources/Apodini/Semantic%20Model%20Builder/ParameterModelBuilder.swift#L84

And the PathModelBuilder equivalent: https://github.com/Apodini/Apodini/blob/806528af4cc6d192e5ab4f527389d221d39bcc12/Sources/Apodini/Semantic%20Model%20Builder/PathModelBuilder.swift#L81